src/js/components/PodSpecView.js

```diff
@@ -0,0 +1,202 @@
+import React from 'react';
+
+import DescriptionList from './DescriptionList';
+import PodSpec from '../structs/PodSpec';
+import PodContainerSpecView from './PodContainerSpecView';
+import Icon from './Icon';
+
+const METHODS_TO_BIND = [
+];
+
+class PodSpecView extends React.Component {
+  constructor() {
+    super(...arguments);
+
+    this.state = {
+    };
+
+    METHODS_TO_BIND.forEach((method) => {
+      this[method] = this[method].bind(this);
+    });
+  }
+
+  getEnvironmentDetails() {
+    let {spec} = this.props;
+    let environment = Object.assign({}, spec.getEnvironment());
+
+    if (Object.keys(environment).length === 0) {
+      return null;
+    }
+
+    Object.keys(environment).forEach(function (key) {
+      let value = environment[key];
+      if (typeof value === 'object' && !Array.isArray(value) &&
+          value !== null) {
+        environment[key] = (
+          <span>
+            <Icon
+              className="icon-margin-right"
+              color="white"
+              family="mini"
+              id="key"
+              size="mini" /> {value.secret}
+          </span>
+        );
+      }
+    });
+
+    return (
+      <div>
+        <h4 className="inverse flush-top">
+          Environment variables
+        </h4>
+        <DescriptionList hash={environment} />
+      </div>
+    );
+  }
+
+  getScalingDetails() {
+    let {spec} = this.props;
+    let scaling = spec.getScaling();
+
+    if (Object.keys(scaling).length === 0) {
+      return null;
+    }
+
+    return (
+      <div>
+        <h4 className="inverse flush-top">
+          Scaling
+        </h4>
+        <DescriptionList hash={scaling} />
+      </div>
+    );
+  }
+
+  getSecretsDetails() {
+    let {spec} = this.props;
+    let secrets = Object.assign({}, spec.getSecrets());
+
+    if (Object.keys(secrets).length === 0) {
+      return null;
+    }
+
+    Object.keys(secrets).forEach(function (key) {
+      let {source} = secrets[key];
+      secrets[key] = source;
+    });
+
+    return (
+      <div>
+        <h4 className="inverse flush-top">
+          Secrets
+        </h4>
+        <DescriptionList hash={secrets} />
+      </div>
+    );
+  }
+
+  getLabelsDetails() {
+    let {spec} = this.props;
+    let labels = spec.getLabels();
+
+    if (Object.keys(labels).length === 0) {
+      return null;
+    }
+
+    return (
+      <div>
+        <h4 className="inverse flush-top">
+          Labels
+        </h4>
+        <DescriptionList hash={labels} />
+      </div>
+    );
+  }
+
+  getGeneralDetails() {
+    let {spec} = this.props;
+    let hash = {
```

Comment:
I think you could do something like this:
```js
let hash = {
  ID: spec.getId(),
  version: spec.getVersion(),
  user: spec.getUser()
};

// Remove falsy values
hash = Object.values(hash).filter(Boolean).reduce(function (memo, key) {
  memo[key] = hash[value];

  return memo;
}, {});
```

(This filter + reduce part could be a promoted to a util function)